transformers/run_eval.py

```diff
@@ -4,47 +4,95 @@
 from transformers import pipeline
 import evaluate
 from normalizer import data_utils
+import time
+from tqdm import tqdm
 
 wer_metric = evaluate.load("wer")
 
 
-def dataset_iterator(dataset):
-    for i, item in enumerate(dataset):
-        yield {**item["audio"], "reference": item["norm_text"]}
-
-
 def main(args):
     asr_pipe = pipeline(
-        "automatic-speech-recognition", model=args.model_id, device=args.device
+        "automatic-speech-recognition", model=args.model_id, device=args.device, batch_size=args.batch_size
     )
 
+    if asr_pipe.model.can_generate():
+        gen_kwargs = {"max_new_tokens": 128}
+        # for multilingual Whisper-checkpoints we see a definitive WER boost by setting the language and task args
+        if (
+            hasattr(asr_pipe.model.generation_config, "is_multilingual")
+            and asr_pipe.model.generation_config.is_multilingual
+        ):
+            gen_kwargs["language"] = "en"
+            gen_kwargs["task"] = "transcribe"
+    else:
+        gen_kwargs = None
+
     dataset = data_utils.load_data(args)
 
     if args.max_eval_samples is not None and args.max_eval_samples > 0:
-        print(f"Subsampling dataset to first {args.max_eval_samples} samples !")
+        print(f"Subsampling dataset to first {args.max_eval_samples} samples!")
         dataset = dataset.take(args.max_eval_samples)
 
     dataset = data_utils.prepare_data(dataset)
 
-    predictions = []
-    references = []
+    def benchmark(batch):
```

Comment:
There are small modifications required to our benchmark function to compute the RTF @nithinraok, mainly:
1. Recording the length of each audio sample
2. Recording the transcription time over the batch